pkgs/by-name/ha/hare/package.nix

```diff
@@ -37,7 +71,14 @@ stdenv.mkDerivation (finalAttrs: {
       src = ./001-tzdata.patch;
       inherit tzdata;
     })
-  ];
+  ] ++ lib.lists.optional enableCrossCompilation
+    # Set paths for cross-compilation toolchains
+    (substituteAll {
+      src = ./002-cross-compilation.patch;
+      inherit (tools.x86_64) x86_64_ld x86_64_as x86_64_cc;
+      inherit (tools.aarch64) aarch64_ld aarch64_as aarch64_cc;
+      inherit (tools.riscv64) riscv64_ld riscv64_as riscv64_cc;
+    });
```

Comment:
maybe it makes more sense to do this substitution in preConfigure, where the logic is currently:
```nix
preConfigure = ''
  ln -s config.example.mk config.mk
'';
```

feels odd to be patching a source file named `example`.

Reply:
Here are the `strings` output after such change on a `Hare` binary built for `x86_64`:

```console
[user@hostname:~/nixpkgs]$ strings result/bin/.hare-wrapped | grep -P '/nix/store/.*/(aarch64|riscv64)'

/nix/store/<hash>/bin/aarch64-unknown-linux-gnu-as
/nix/store/<hash>/bin/aarch64-unknown-linux-gnu-cc
/nix/store/<hash>/bin/aarch64-unknown-linux-gnu-ld
/nix/store/<hash>/bin/riscv64-unknown-linux-gnu-as
/nix/store/<hash>/bin/riscv64-unknown-linux-gnu-cc
/nix/store/<hash>/bin/riscv64-unknown-linux-gnu-ld


```